analyses/fusion_filtering/04-project-specific-filtering.Rmd

```diff
@@ -0,0 +1,269 @@
+# fusion_summary<-left_join(fusion_caller.summary,fusion_read.summary,by=c("Sample","FusionName"))
+#  
+
+# remove fusion within local rearrangement
+fusion_calls <- fusion_calls %>%
+  # remove local rearrangement/adjacent genes
+  dplyr::filter(!grepl("LOCAL_REARRANGEMENT|LOCAL_INVERSION",annots))
+
+#to add aggregated caller from fusion_caller.summary
+fusion_calls<-fusion_calls %>% 
+  dplyr::filter(Fusion_Type != "other") %>% dplyr::select(-Caller,-annots) %>%
+  left_join(fusion_caller.summary,by=(c("Sample","FusionName","Fusion_Type"))) %>%
+  dplyr::select(-JunctionReadCount,-SpanningFragCount,-Confidence,-LeftBreakpoint,-RightBreakpoint) %>% unique()
+
+#merge with histology file
+fusion_calls<-merge(fusion_calls,clinical,by.x="Sample",by.y="Kids_First_Biospecimen_ID")
+
+
+
+
+
+
+```
+
+
+```{r}
+# Gene fusion should be in-frame
+# AND
+#
+# 1. Called by at least n callers
+fusion_calls.summary <- fusion_calls %>%
+  dplyr::filter(caller.count >= numCaller) %>%
+  unique() %>%
+  mutate(note=paste0("Called by",numCaller, "callers")) %>%
+  as.data.frame()
+
+# OR
+# 2. Found in at least n samples in each group
+sample.count <- fusion_calls %>%
+  dplyr::filter(Fusion_Type != "other") %>%
+  dplyr::select(FusionName, Sample, !!as.name(group),-Fusion_Type) %>%
+  unique() %>%
+  group_by(FusionName, !!as.name(group)) %>%
+  dplyr::mutate(sample.count = n(),Sample = toString(Sample)) %>%
+  dplyr::filter(sample.count > numSample) %>%
+  unique() %>%
+  mutate(note=paste0("Found in atleast ",numSample, " samples in a group")) %>%
+  as.data.frame()
+
+
+```
+
+
+
+```{r}
+#filter QCGeneFiltered_filtFusion to keep recurrent fusions from above sample.count and fusion_calls.summary
+
+QCGeneFiltered_recFusion<-fusion_calls %>%
+  dplyr::filter(FusionName %in% unique(c(sample.count$FusionName,fusion_calls.summary$FusionName)))
+
+```
+
+
+
+
+
+```{r}
+# remove fusions that are in > numGroup
+group.count <- fusion_calls %>%
+  dplyr::select(FusionName, !!as.name(group)) %>%
+  unique() %>%
+  group_by(FusionName) %>%
+  dplyr::mutate(group.ct = n(),Sample = toString(!!(as.name(group)))) %>%
+  dplyr::filter(group.ct >numGroup)
+
+
+# remove multi-fused genes
+fusion_recurrent5_per_sample <- fusion_calls  %>%
+  # We want to keep track of the gene symbols for each sample-fusion pair
+  dplyr::select(Sample, FusionName, Gene1A, Gene1B, Gene2A, Gene2B) %>%
+  # We want a single column that contains the gene symbols
+  tidyr::gather(Gene1A, Gene1B, Gene2A, Gene2B,
+                key = gene_position, value = GeneSymbol) %>%
+  # Remove columns without gene symbols
+  dplyr::filter(GeneSymbol != "") %>%
+  dplyr::arrange(Sample, FusionName) %>%
+  # Retain only distinct rows
+  dplyr::distinct() %>%
+  group_by(Sample,GeneSymbol) %>% 
+  dplyr::summarise(Gene.ct = n()) %>% 
+  dplyr::filter(Gene.ct>limitMultiFused) %>%
+  mutate(note=paste0("multfused " ,limitMultiFused, " times per sample"))
+
+
+
+```
+
+
+```{r}
+
+
+outputfolder<-"results"
```

Comment:
I would move this into `params` because that is where I would expect this information to be based on the rest of the notebook. I don't think you need the working directory of this chunk to be in `results`.

Reply:
done